sycl/doc/DynamicLinkingTestPlan.md

```diff
@@ -0,0 +1,179 @@
+# Test plan for dynamic linking of device code
+
+This is a test plan for device code dynamic linking feature described in
+https://github.com/intel/llvm/blob/sycl/sycl/doc/SharedLibraries.md document.
+
+## Testing scope
+
+### Device & BE coverage
+
+All of the tests described below are performed on Intel CPU and GPU devices.
+For GPU device both OpenCL and L0 backends are used.
+
+**NOTE**: The feature isn't supported on FPGA device.
+
+### AOT & JIT compilation
+
+All the tests are compiled using AOT, JIT and mixed (AOT with JIT) compilation.
```

Comment:
What do you mean by mixed `AOT with JIT` compilation? Is it like one source is compiled with JIT and another with AOT within the application/shared library or it is like application is compiled in JIT mode, whilst library is AOT and vice-versa? Should we clarify that?

Reply:
I actually meant the case when one binary is built with one type of compilation and other with another type of compilation. For example library built for JIT mode and application built in AOT mode.
We don't have support to compile files in different modes inside a single binary. All files can be compiled in the same mode, or user can request to compile for both modes and for several devices, but then each file will be compiled in all those modes.

Reply:
Since we don't have support for mixed compilation inside one binary, should we really add this clarification?

Reply:
> Since we don't have support for mixed compilation inside one binary

Why is that the case? My understanding is that I should be able to compile a one `.cpp` into a fat object with JIT target and another `.cpp` to fat object with AOT target, then link them together and launch an app. At least if there are no dependencies between those two object files it should work fine. But I might be missing something.

> should we really add this clarification?

I would still vote for this, because (a) things could change and (b) it might not be clear for everyone what was the intent like it wasn't clear for me

Reply:
> Why is that the case? My understanding is that I should be able to compile a one .cpp into a fat object with JIT target and another .cpp to fat object with AOT target, then link them together and launch an app.

AFAIK it is not possible to do this thing. AOT compilers (as well as SPIR-V translator) are ran after link stage, so all code is compiled using the same mode. But ok, I'll add this clarification.